### DISCUSSION TITLE
Any idea why when connecting aqualinkD it causes aqualink to shutdown

### DISCUSSION THREAD

**Discussion by sjames99** [aqualinkd/AqualinkD#320] [Q&A] [answered]
When I get everything connected and power up aqualinkd, it will read the config and start the daemon.  Then aqualink immediately shuts down, as soon as I unplug the aqualinkd device it immediately comes back online.  I get nothing in the log, last thing posted is the clock sync.  Then it is dead in the water, all pool equipment powers down until it is unplugged.

**Comment by sfeakes** [comment]
Sounds like AqualinkD and Aqualink are using the same ID.  Did you run serial-logger to check what IDs are free/being used?  (And make sure aqualink wasnt asleep when doing that)

**Reply by sjames99** [reply]
I will try the serial-logger, if I stop aqualinkd before connecting I should be able to execute it.  As it seems stable as long as not connected to aqualink.

**Reply by sfeakes** [reply]
You can't run serial-logger and aqualinkd at the same time, since they both need access to the USBRS485 converter. 

By aqualink, I am assuming you are talking about Jandy's Aqualink interfaces (like iAqualink or Aqualink Onetouch etc)

**Reply by sjames99** [reply]
Sorry, I meant before connecting the serial interface to the jandy panel.  It crashes so fast I wouldn’t have to connect and then stop the service.  But I should be able power up, stop the service and then run the serial logger.

**Reply by sjames99** [reply]
So serial_logger just hangs without returning anything so something must be wrong with my serial port connection.  Need to look deeper at that.

**Reply by sjames99** [reply]
Re-did my serial connection and successfully tested it with a loopback.  I changed the ID to 0x00b, aqualinkd doesn't crash immediately.  However, the jandy panel still powers down a minute or so after serial connection.  I did verify my wiring is correct.  Once I disconnect the serial connection, the jandy panel immediately returns and all equipment is back operational.

Serial_logger just hangs at reading information.  
I did notice that stopping aqualinkd takes about 1.5 minutes before it shuts down, in looking at the logs I noticed RS Serial: Didn't find valid blocking serial port file descripter after the shutdown command.  

I am kind of out of ideas.

**Comment by sfeakes** [comment, marked as answer]
In your first post, I thought you were saying that AqualinkD was working correctly, but a Jandy interface was shutting down when running. From your last reply that can't be the case that AqualinkD is running correctly.  So my initial comments wouldn't help.

From your last reply, it sounds like the candy control panel is shutting down?  If that's the case, your wiring can't be correct.  Let's go back to the basics.
You have a Jandy RS control panel?
You are using a RS4852USB serial interface?
On RS4852USB interface, you are ONLY connecting 2 wires, (Data+ and Data-) to the RS control panel?

This message `RS Serial: Didn't find valid blocking serial port file descripter` means aqualinkd did NOT find the RS4852USB serial interface. 

You mention testing wiring by a "loop back", what do you mean by that, to my knolage a loop back test is serial only, and you have a USB2RS485 adapter (no serial pins you can actually access).
Have you read the wiki, and followed the wiring the control panel section, [Link](https://github.com/sfeakes/AqualinkD/wiki#hardware---connecting-rs485-to-control-panel) and followed the initial test [Link](https://github.com/sfeakes/AqualinkD/wiki#usb-device-on-raspberrypi--linux-computer)

**Reply by sjames99** [reply]
I've tried both, pins and the RS4852USB.  Both will show as devices and I can test them.  However, once connected to the jandy panel, the panel will shutdown.  I rechecked the wiring today and everything looks good including re-wiring and removing some devices to slim down the channel as much as possible.

The initial test for the USB device is:
goldfish@AqualinkD:~$ dmesg | grep usb
[    0.000000] Kernel command line: root=UUID=551c3a76-a854-4201-b126-1cef2a2949                                                                                                             64 rootwait rootfstype=ext4 splash=verbose console=ttyS0,115200 console=tty1 con                                                                                                             soleblank=0 loglevel=1 ubootpart=bb229d64-01 usb-storage.quirks=0x2537:0x1066:u,                                                                                                             0x2537:0x1068:u   cgroup_enable=memory swapaccount=1
[    0.654425] usbcore: registered new interface driver usbfs
[    0.654459] usbcore: registered new interface driver hub
[    0.654487] usbcore: registered new device driver usb
[    1.617113] usbcore: registered new interface driver usb-storage
[    1.623438] usbcore: registered new interface driver usbhid
[    1.623444] usbhid: USB HID core driver
[    1.747539] ehci-platform 5200000.usb: EHCI Host Controller
[    1.747578] ehci-platform 5200000.usb: new USB bus registered, assigned bus n                                                                                                             umber 1
[    1.747696] ehci-platform 5200000.usb: irq 290, io mem 0x05200000
[    1.747932] ehci-platform 5310000.usb: EHCI Host Controller
[    1.747950] ehci-platform 5310000.usb: new USB bus registered, assigned bus n                                                                                                             umber 2
[    1.748028] ehci-platform 5310000.usb: irq 291, io mem 0x05310000
[    1.748071] ehci-platform 5311000.usb: EHCI Host Controller
[    1.748089] ehci-platform 5311000.usb: new USB bus registered, assigned bus n                                                                                                             umber 3
[    1.748167] ehci-platform 5311000.usb: irq 292, io mem 0x05311000
[    1.748824] ohci-platform 5200400.usb: Generic Platform OHCI controller
[    1.748843] ohci-platform 5200400.usb: new USB bus registered, assigned bus n                                                                                                             umber 4
[    1.748883] ohci-platform 5310400.usb: Generic Platform OHCI controller
[    1.748902] ohci-platform 5310400.usb: new USB bus registered, assigned bus n                                                                                                             umber 5
[    1.748920] usb_phy_generic usb_phy_generic.4.auto: supply vcc not found, usi                                                                                                             ng dummy regulator
[    1.748921] ohci-platform 5200400.usb: irq 293, io mem 0x05200400
[    1.748967] ohci-platform 5310400.usb: irq 294, io mem 0x05310400
[    1.749035] usb_phy_generic usb_phy_generic.4.auto: dummy supplies not allowe                                                                                                             d for exclusive requests
[    1.749184] ohci-platform 5311400.usb: Generic Platform OHCI controller
[    1.749201] ohci-platform 5311400.usb: new USB bus registered, assigned bus n                                                                                                             umber 6
[    1.749303] ohci-platform 5311400.usb: irq 295, io mem 0x05311400
[    1.760460] ehci-platform 5200000.usb: USB 2.0 started, EHCI 1.00
[    1.760708] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bc                                                                                                             dDevice= 6.01
[    1.760722] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.760731] usb usb1: Product: EHCI Host Controller
[    1.760739] usb usb1: Manufacturer: Linux 6.1.31-sun50iw9 ehci_hcd
[    1.760747] usb usb1: SerialNumber: 5200000.usb
[    1.776451] ehci-platform 5310000.usb: USB 2.0 started, EHCI 1.00
[    1.776681] usb usb2: New USB device found, idVendor=1d6b, idProduct=0002, bc                                                                                                             dDevice= 6.01
[    1.776692] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.776701] usb usb2: Product: EHCI Host Controller
[    1.776707] usb usb2: Manufacturer: Linux 6.1.31-sun50iw9 ehci_hcd
[    1.776714] usb usb2: SerialNumber: 5310000.usb
[    1.792452] ehci-platform 5311000.usb: USB 2.0 started, EHCI 1.00
[    1.792685] usb usb3: New USB device found, idVendor=1d6b, idProduct=0002, bc                                                                                                             dDevice= 6.01
[    1.792698] usb usb3: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.792708] usb usb3: Product: EHCI Host Controller
[    1.792717] usb usb3: Manufacturer: Linux 6.1.31-sun50iw9 ehci_hcd
[    1.792727] usb usb3: SerialNumber: 5311000.usb
[    1.812731] usb usb5: New USB device found, idVendor=1d6b, idProduct=0001, bc                                                                                                             dDevice= 6.01
[    1.812753] usb usb5: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.812769] usb usb5: Product: Generic Platform OHCI controller
[    1.812782] usb usb5: Manufacturer: Linux 6.1.31-sun50iw9 ohci_hcd
[    1.812794] usb usb5: SerialNumber: 5310400.usb
[    1.814128] usb usb4: New USB device found, idVendor=1d6b, idProduct=0001, bc                                                                                                             dDevice= 6.01
[    1.814150] usb usb4: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.814166] usb usb4: Product: Generic Platform OHCI controller
[    1.814179] usb usb4: Manufacturer: Linux 6.1.31-sun50iw9 ohci_hcd
[    1.814191] usb usb4: SerialNumber: 5200400.usb
[    1.815346] usb usb6: New USB device found, idVendor=1d6b, idProduct=0001, bc                                                                                                             dDevice= 6.01
[    1.815359] usb usb6: New USB device strings: Mfr=3, Product=2, SerialNumber=                                                                                                             1
[    1.815369] usb usb6: Product: Generic Platform OHCI controller
[    1.815376] usb usb6: Manufacturer: Linux 6.1.31-sun50iw9 ohci_hcd
[    1.815384] usb usb6: SerialNumber: 5311400.usb
[    2.344483] usb 4-1: new full-speed USB device number 2 using ohci-platform
[    2.583529] usb 4-1: New USB device found, idVendor=0403, idProduct=6001, bcd                                                                                                             Device= 6.00
[    2.583558] usb 4-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    2.583578] usb 4-1: Product: USB-RS485 Cable
[    2.583593] usb 4-1: Manufacturer: FTDI
[    2.583608] usb 4-1: SerialNumber: FT7UHZ8T
[    9.872211] usbcore: registered new interface driver usbserial_generic
[    9.872263] usbserial: USB Serial support registered for generic
[    9.935657] usbcore: registered new interface driver ftdi_sio
[    9.935711] usbserial: USB Serial support registered for FTDI USB Serial Devi                                                                                                             ce
[    9.935983] usb 4-1: Detected FT232R
[    9.941222] usb 4-1: FTDI USB Serial Device converter now attached to ttyUSB0
[   90.047748] usb 4-1: USB disconnect, device number 2
[   90.528936] usb 4-1: new low-speed USB device number 3 using ohci-platform
[   90.720913] usb 4-1: device descriptor read/64, error -62
[   91.020922] usb 4-1: device descriptor read/64, error -62
[   91.316913] usb 4-1: new full-speed USB device number 4 using ohci-platform
[   91.555519] usb 4-1: New USB device found, idVendor=0403, idProduct=6001, bcd                                                                                                             Device= 6.00
[   91.555568] usb 4-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[   91.555588] usb 4-1: Product: USB-RS485 Cable
[   91.555604] usb 4-1: Manufacturer: FTDI
[   91.555620] usb 4-1: SerialNumber: FT7UHZ8T
[   91.561937] usb 4-1: Detected FT232R
[   91.568185] usb 4-1: FTDI USB Serial Device converter now attached to ttyUSB0
[   92.549420] usb 4-1: USB disconnect, device number 4
[  101.548884] usb 4-1: new full-speed USB device number 5 using ohci-platform
[  101.787923] usb 4-1: New USB device found, idVendor=0403, idProduct=6001, bcd                                                                                                             Device= 6.00
[  101.787956] usb 4-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  101.787977] usb 4-1: Product: USB-RS485 Cable
[  101.787993] usb 4-1: Manufacturer: FTDI
[  101.788009] usb 4-1: SerialNumber: FT7UHZ8T
[  101.794890] usb 4-1: Detected FT232R
[  101.800636] usb 4-1: FTDI USB Serial Device converter now attached to ttyUSB0

goldfish@AqualinkD:~$ sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev                                                                                                             /ttyUSB0'


The echo command does nothing, similar to the serial_logger.  This behavior is the same with pins and the adapter.  I might just start over and rebuild and instead of trying to throw different things at it.

**Comment by sfeakes** [comment]
If the command `sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev /ttyUSB0'` didn't work, then their is no chance AqualinkD will work. That is showing that nothing if being read on the RS485 protocol.

I'd suggest wiring the adapter in to the RS485 bus WITHOUT any power (ie don't connect it to the raspberry pi), as just use data+/- lines no others.  If the control panel is still working, then connect it to the raspberry Pi (without AqualinkD running), and check control panel again, as that should power up the adapter.  If either of those fail, it's either wiring or a bad adapter.  If they work, then try the `stty` command again and see if you get output.  If you do, and panel is still operational than start AqualinkD.  But I doubt you'll get to that stage.

BTW, what USB2RS485 adapter are you using?

**Reply by sjames99** [reply]
It is the FTDI one linked in the wiki. 

I will get that test a try.  It has to be something simple just haven’t been able to narrow down.

**Reply by sfeakes** [reply]
On that adapter, you should only have 2 wires connected to the RS485 buss and all other wires left exposed (not touching anything) for the initial test/setup.

**Reply by sjames99** [reply]
Finally got something:

Jul 28 14:57:58 aqualinkd aqualinkd[651]: AqualinkD: Aqualink daemon looks like serial error, resetting.
Jul 28 14:57:58 aqualinkd aqualinkd[651]: RS Serial: Port /dev/ttyUSB0 low latency mode is set
Jul 28 14:57:58 aqualinkd aqualinkd[651]: RS Serial: Serial read bad Jandy checksum, ignoring
Jul 28 14:57:58 aqualinkd aqualinkd[651]: RS Serial: Read  Jandy   packet BAD PACKET To 0x48 of type            Probe | HEX: 0x10|0x02|0x48|0x00|0x10|0x03|
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Serial read bad Jandy checksum, ignoring
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Read  Jandy   packet BAD PACKET To 0x00 of type   Unknown '0x73' | HEX: 0x10|0x02|0x00|0x73|0x10|0x02|0x3e|0x0
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Serial read bad Jandy checksum, ignoring
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Read  Jandy   packet BAD PACKET To 0x93 of type  PDA HlightChars | HEX: 0x10|0x02|0x93|0x10|0x02|0x50|0x73|0x1
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Serial read bad Jandy checksum, ignoring
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Read  Jandy   packet BAD PACKET To 0x70 of type            Probe | HEX: 0x10|0x02|0x70|0x00|0x00|0x10|0x03|
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Serial read bad Jandy checksum, ignoring
Jul 28 14:57:59 aqualinkd aqualinkd[651]: RS Serial: Read  Jandy   packet BAD PACKET To 0x00 of type   Unknown '0x73' | HEX: 0x10|0x02|0x00|0x73|0x10|0x02|0x3e|0x0

Assumption is the checksums are the result of a wiring issue.  But progress none the less.

Serial logger output:
goldfish@aqualinkd:~/software/AqualinkD $ sudo ./release/serial_logger /dev/ttyU                                                         SB0
AqualinkD serial_logger V2.3
Notice:  RS Serial: Port /dev/ttyUSB0 low latency mode is set
Notice:  RS Serial: Logging serial information!
Please wait.Warning: RS Serial: Serial read bad Jandy checksum, ignoring
Warning: RS Serial: Read  Jandy   packet BAD PACKET To 0x70 of type  LXi heater                                                          ping | HEX: 0x10|0x02|0x70|0x0c|0x00|0x00|0x00|0x00|0x8e|0x00|0x10|0x02|0x00|0x0                                                         1|0x00|0x00|0x13|0x10|0x03|
Warning: RS Serial: Serial read bad Jandy checksum, ignoring
Warning: RS Serial: Read  Jandy   packet BAD PACKET To 0x00 of type            Probe | HEX: 0x10|0x02|0x00|0x00|0x00|0x00|0x04|0x34|0xf8|0x10|0x02|0x38|0x00|0x4a|0x10|0x03|
-

Notice:  RS Serial: RS485 interface received 600 packets in 24 seconds (~25.00 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   :  RS-4 Combo
Notice:  RS Serial: Jandy Control Panel Version :  REV T.2g
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0x70 is in use    <-- Unknown Device
Notice:  RS Serial: ID 0x60 is not used  <-- can use for Aqualinkd (PDA mode only)
Notice:  RS Serial: ID 0x08 is not used  <-- can use for Aqualinkd
Notice:  RS Serial: ID 0x09 is not used  <-- can use for Aqualinkd
Notice:  RS Serial: ID 0x0a is not used  <-- can use for Aqualinkd
Notice:  RS Serial: ID 0x0b is not used  <-- can use for Aqualinkd
Notice:  RS Serial: ID 0x40 is not used  <-- can use for Aqualinkd (Extended Device ID)
Notice:  RS Serial: ID 0x41 is not used  <-- can use for Aqualinkd (Extended Device ID)
Notice:  RS Serial: ID 0x42 is not used  <-- can use for Aqualinkd (Extended Device ID)
Notice:  RS Serial: ID 0x79 is in use    <-- Jandy VSP ePump
Notice:  RS Serial: ID 0x43 is not used  <-- can use for Aqualinkd (Extended Device ID)
Notice:  RS Serial: ID 0x78 is in use    <-- Jandy VSP ePump
Notice:  RS Serial: ID 0x30 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x31 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x32 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x33 is not used  <-- can use for Aqualinkd (Prefered Extended Device ID)
Notice:  RS Serial: ID 0x48 is not used  <-- can use for Aqualinkd (RSSA ID)
Notice:  RS Serial: ID 0x50 is in use    <-- Salt Water Generator (Aquarite mode)
Notice:  RS Serial:

Notice:  RS Serial: Suggested aqualinkd.conf values
Notice:  RS Serial: -------------------------
Notice:  RS Serial: panel_type =  RS-4 Combo
Notice:  RS Serial: device_id = 0x08
Notice:  RS Serial: rssa_device_id = 0x48
Notice:  RS Serial: extended_device_id = 0x30
Notice:  RS Serial: -------------------------

**Reply by sjames99** [reply]
It was the wiring.  I started over and examined the wire and notice that the Data+ wire was actually broken in the sleeve.  So the connect was intermittent and the large number of checksum errors going back forth would cause the panel to shut down.  Once the FTDI connector was removed the panel would come right back online.
